target/chip.py

```diff
@@ -0,0 +1,50 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""OpenTitan chip utility functions."""
+import time
+from subprocess import PIPE, Popen
+
+
+class Chip():
+    """Class for the discrete chip. Initializes OpenTitan with the provided
+    firmware & provides helper functions.
+    """
+    def __init__(self, firmware, opentitantool_path):
+        self.firmware = firmware
+        self.opentitantool = opentitantool_path
+        self._initialize_chip()
+
+    def _initialize_chip(self):
+        """Initializes the chip."""
+        # Flash the chip using the opentitantool with the provided firmware.
+        flash_process = Popen([self.opentitantool, "--rcfile=",
+                               "--interface=hyper310",
+                               "--exec", "transport init",
+                               "--exec", "bootstrap " + self.firmware, "no-op"],
+                              stdout=PIPE, stderr=PIPE)
+        flash_process.communicate()
+        rc = flash_process.returncode
+        if rc != 0:
+            raise RuntimeError('Error: Failed to flash OpenTitan chip.')
+        else:
+            # Wait until chip finished booting.
+            time.sleep(1)
```

Comment:
How sure are we that 1s is enough to finish booting?

Reply:
Good point - I've made some experiments and I think that 1s should be more than enough. For FI experiments, we might even reduce it to 500 or 750ms. I am changing the code such that the user can specify the boot delay.